lib/zlib.js

```diff
@@ -224,14 +224,32 @@ class Zlib extends Transform {
     this._hadError = false;
 
     var level = constants.Z_DEFAULT_COMPRESSION;
-    if (typeof opts.level === 'number') level = opts.level;
+    if (typeof opts.level === 'number' &&
+        Number.isFinite(opts.level)) {
```

Comment:
Nit: isn't `Number.isFinite()` sufficient here? I think it would be easier to read without the `typeof` check.